weave-js/src/components/Tag/Tag.tsx

```diff
@@ -103,11 +103,21 @@ export const RemovableTag: FC<RemovableTagProps> = ({
   Wrapper = Tailwind,
 }) => {
   const labelRef = useRef<HTMLParagraphElement>(null);
-  const isTooltipEnabled = isTagLabelTruncated(labelRef);
+  const [isTruncated, setIsTruncated] = useState(false);
+
+  useEffect(() => {
+    const checkTruncation = () => {
+      if (labelRef.current) {
+        setIsTruncated(isTagLabelTruncated(labelRef));
+      }
+    };
+    checkTruncation();
```

Comment:
[nit] you could even remove this function since you aren't using it elsewhere 😄 
```suggestion
    if (labelRef.current) {
      setIsTruncated(isTagLabelTruncated(labelRef));
    }
```

Reply:
Great suggestion! Thank you for bringing it up. Done.